src/app/pages/categories/management/categories-editor/pages-categories-editor.component.ts

```diff
@@ -0,0 +1,97 @@
+import { Component, Inject, OnInit } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
+import { checkInputError } from 'src/app/common/utils/forms/common-utils-forms-form-utils';
+import { WalletTransactionType } from 'src/app/domains/transactions/domains.transactions.constants';
+import { PagesTransactionCategoriesService } from '../../pages-transaction-categories.service';
+import { TransactionCategoryFull } from '../../transaction-category-full.model';
+import { TransactionCategory } from '../../transaction-category.model';
+import { ITransactionCategoryModalFormType } from './pages-categories-editor.types';
+
+@Component({
+  selector: 'app-pages-categories-editor',
+  templateUrl: './pages-categories-editor.component.html',
+  styleUrls: [ './pages-categories-editor.component.scss' ],
+})
+export class PagesCategoriesEditorComponent implements OnInit {
+  public selectTransactionsTypes: Record<string, WalletTransactionType> = {
+    'Income': WalletTransactionType.Income,
+    'Expense': WalletTransactionType.Expense,
+  };
+
+  public form!: FormGroup<ITransactionCategoryModalFormType>;
+
+  public transactionTypeDisabledLoading = true;
+
+  private category: TransactionCategory | null = null;
```

Comment:
Ah damn it, now when I see this we can simplify this even more. From the list to the editor we should only pass `type` and optional `categoryId`. And in editor:
- create empty `TransactionCategoryFull` if we are creating new category
- GET by ID from BE json and create `TransactionCategoryFull` if we are updating one

That way we don't have to have here two classes for category.